File: backend/src/models/event.go
package models

import "time"

type Event struct {
	EventId   int       `json:"eventId" form:"eventId" binding:"required"`
	GroupId   int       `json:"groupId" form:"groupId"`
	EventName string    `json:"eventName" form:"eventName"`
	StartTime time.Time `json:"startTime" form:"startTime"`
	EndTime   time.Time `json:"endTime" form:"endTime"`
}

File: apps/schedulii_api/src/models/event.rs
use serde::{Deserialize, Serialize};
use sqlx::types::chrono::{DateTime, Utc};
use sqlx::FromRow;

#[derive(Debug, FromRow, Serialize, Deserialize)]
pub struct Event {
    pub event_id: i32,
    pub event_name: String,
    pub start_time: DateTime<Utc>,
    pub end_time: DateTime<Utc>,
    pub location: String,
}

#[derive(Debug, FromRow, Serialize, Deserialize)]
pub struct NewEvent {
    pub event_name: String,
    pub start_time: DateTime<Utc>,
    pub end_time: DateTime<Utc>,
    pub location: String,
}
